include/circt/Support/Passes.h

```diff
@@ -28,13 +32,22 @@ class VerbosePassInstrumentation : public mlir::PassInstrumentation {
   int level = 0;
   const char *toolName;
 
+  void emitToolNameAndTimestamp(llvm::raw_fd_ostream &os, TimePoint &now) {
+    auto time = std::chrono::system_clock::to_time_t(now);
+    std::stringstream ss;
+    ss << std::put_time(std::localtime(&time), "%Y-%m-%d %X");
```

Comment:
Can this include `%Z` to show the time zone, too? These logs _may_ be confusing or require users to guess at when something happened if they are inspecting a log from somebody else that is using a different locale.

Reply:
Interesting.  Maybe just print UTC?  I don't wanna wrangle timezones.

Especially when run in containers, namespaces, so on, these probably should not be expected to reflect a particular locale or even comparable absolute time -- basically don't trust system clocks (might not even be monotonic!) or use them for "happened-before" relations across different clocks (different machines / environments).

That said, not 100% what the goal is here re:printing the system clock occasionally, especially as part of our tool?
Would just printing the system clock at the start communicate the same information?

Reply:
The request was to print timestamps along with messages so users can know "when the last step started" easily. I don't have a strong opinion about the format or timezone. Maybe it makes sense to make the format itself CLI option?